tests/mock_network/utils.rs

```diff
@@ -491,60 +558,30 @@ pub fn add_mature_nodes(
             add_node_to_section(env, nodes, &sub_prefix0);
             remaining0 -= 1;
         }
-    }
 
-    poll_and_resend(nodes);
-
-    // Remove 16 mature nodes to trigger 16 age increments.
-    info!("Removing {} mature nodes", remove_count);
-    for _ in 0..remove_count {
-        // Note: removing only elders for simplicity. Also making sure we don't remove any of the
-        // last `count0 + count1` nodes.
-        remove_elder_from_section_in_range(nodes, &prefix, 0..nodes.len() - count0 - count1);
-        poll_and_resend(nodes);
+        let i = count0 + count1 - remaining0 - remaining1;
+        if i % MAX_SIMULTANEOUS_JOINS == 0 {
+            poll_until_last_nodes_joined(env, nodes, first_index);
+            first_index = nodes.len();
+        }
     }
 
-    // Count the number of nodes in each sub-prefix and verify they are as expected.
-    let actual_count0 = nodes_with_prefix(nodes, &sub_prefix0).count();
-    assert_eq!(actual_count0, count0);
-
-    let actual_count1 = nodes_with_prefix(nodes, &sub_prefix1).count();
-    assert_eq!(actual_count1, count1);
+    poll_until_last_nodes_joined(env, nodes, first_index);
 }
 
-fn poll_until_split(nodes: &mut [TestNode], prefix: &Prefix<XorName>) {
-    let sub_prefix0 = prefix.pushed(false);
-    let sub_prefix1 = prefix.pushed(true);
-
-    poll_and_resend_with_options(
-        nodes,
-        PollOptions::default().continue_if(move |nodes| {
-            let mut pending = nodes
-                .iter()
-                .filter(|node| {
-                    (sub_prefix0.matches(node.name()) && *node.our_prefix() != sub_prefix0)
-                        || (sub_prefix1.matches(node.name()) && *node.our_prefix() != sub_prefix1)
-                })
-                .map(|node| node.name())
-                .peekable();
-
-            if pending.peek().is_some() {
-                debug!("Pending split: {}", pending.format(", "));
-                true
-            } else {
-                false
-            }
-        }),
-    )
+fn poll_until_last_nodes_joined(env: &Environment, nodes: &mut [TestNode], first_index: usize) {
+    poll_until(env, nodes, |nodes| {
+        all_nodes_joined(nodes, first_index..nodes.len())
+    })
 }
 
 // -----  Small misc functions  -----
 
 /// Sorts the given nodes by their distance to `name`. Note that this will call the `name()`
 /// function on them which causes polling, so it calls `poll_all` to make sure that all other
 /// events have been processed before sorting.
-pub fn sort_nodes_by_distance_to(nodes: &mut [TestNode], name: &XorName) {
-    let _ = poll_all(nodes); // Poll
+pub fn sort_nodes_by_distance_to(env: &Environment, nodes: &mut [TestNode], name: &XorName) {
+    poll_all(env, nodes); // Poll
```

Comment:
the comment tail could be removed

Reply:
I'll remove the whole `poll_all` call as it shouldn't be needed.